### PR TITLE
#2561 add name and id in collection history

### LINKED ISSUE #2561: Add the name of the item to the collection history
Add the name of the item to the collection history, once it would be available. 
Right now making this issue name was not available.

_Originally posted by @yangwao in https://github.com/kodadot/nft-gallery/issues/2552#issuecomment-1069577608_

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### What's new?

- [x] PR closes #2561 
- [x] add name and id in collection history through graphql

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged the recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

https://user-images.githubusercontent.com/734428/158996881-4e396c85-7b23-46ee-a327-f05233a6fafe.mov

### CODE REVIEW

**Comment by yangwao** [comment]
ooo briliant! 

image

File: components/rmrk/Gallery/History.vue [hunk 1]
```diff
@@ -127,7 +127,7 @@ const components = {
 
 type TableRow = {
   Type: string
-  Item?: string // only in collection
+  Item?: object // only in collection
```
Comment (vikiival) [inline components/rmrk/Gallery/History.vue:130]:
```suggestion
  Item?: TableRowItem // only in collection
```
`object` is not a proper type 👀 

Please create type like 
```ts
type TableRowItem = {
  id: string
  name: string
}
```

**Review by vikiival — CHANGES_REQUESTED** [review]
Really nice contribution!

Just fix the type and we are good to go!

**Comment by yangwao** [comment]
@preschian 👀 😇

File: components/rmrk/Gallery/History.vue (hunk 1 above)
Comment (preschian) [reply components/rmrk/Gallery/History.vue]: thanks for the review, updated the Type for TableRowItem 🙏🏻

**Review by vikiival — APPROVED** [review]
OKi

**Comment by yangwao** [comment]
pay 100 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 143.16 USD/KSM ~ 0.699 $KSM
🧗 [DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM](https://kusama.subscan.io/account/DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)
🔗 [0x0b16b1e58482f3f4d650aa64099f7062aae956edb0d06da6324ccdb45b5eae88](https://kusama.subscan.io/extrinsic/0x0b16b1e58482f3f4d650aa64099f7062aae956edb0d06da6324ccdb45b5eae88)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**